### PR TITLE
[project-base] Redesign store list

### PR DESCRIPTION
**Pull request by henzigo**
#### Description, the reason for the PR
Store list is redesigned based on internal design

#### Fixes issues
... 

#### Check the appropriate checkboxes below, please

- [ ] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [ ] New feature 
- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

## Summary by CodeRabbit

- **New Features**
	- Enhanced store opening hours representation with a more descriptive status system.
	- Introduced a new `StoreOpeningStatusEnum` to indicate various operational states (OPEN, CLOSED, OPEN_SOON, CLOSED_SOON).
	- Added new fields for store opening status in various GraphQL queries and fragments.
	- Implemented a new `StoreList` component for improved navigation and display of store details.

- **Bug Fixes**
	- Resolved inconsistencies in store opening status representation across various components and queries.

- **Documentation**
	- Updated GraphQL schema documentation to reflect new status fields and enums for store opening hours.

----
:globe_with_meridians: Live Preview:
  - https://jg-redesign-store-list.odin.shopsys.cloud
  - https://cz.jg-redesign-store-list.odin.shopsys.cloud

### CODE REVIEW

File: project-base/storefront/components/Blocks/OpeningHours/OpeningStatus.tsx [hunk 1]
```diff
@@ -1,24 +1,43 @@
 import { Flag } from 'components/Basic/Flag/Flag';
+import { TypeStoreOpeningStatusEnum } from 'graphql/types';
 import useTranslation from 'next-translate/useTranslation';
 import { twMergeCustom } from 'utils/twMerge';
 
-export const OpeningStatus: FC<{ isOpen: boolean; isDynamic?: boolean }> = ({
-    isOpen,
+export const OpeningStatus: FC<{ status: TypeStoreOpeningStatusEnum; isDynamic?: boolean }> = ({
```
Comment (chlebektomas) [inline project-base/storefront/components/Blocks/OpeningHours/OpeningStatus.tsx]: To maintain consistency across the codebase, I recommend outsourcing props like this: 
`type OpeningStatusProps = { status: TypeStoreOpeningStatusEnum; isDynamic?: boolean; }`.

File: project-base/storefront/components/Blocks/OpeningHours/OpeningStatus.tsx [hunk 2]
```diff
@@ -1,24 +1,43 @@
...
-                isOpen ? 'bg-openingStatusOpen' : 'bg-openingStatusClosed',
+                status === TypeStoreOpeningStatusEnum.Open && 'bg-openingStatusOpen',
+                status === TypeStoreOpeningStatusEnum.Closed && 'bg-openingStatusClosed',
+                (status === TypeStoreOpeningStatusEnum.OpenSoon || status === TypeStoreOpeningStatusEnum.ClosedSoon) &&
+                    'bg-openingStatusOpenToday',
                 className,
             )}
         >
-            {isOpen ? t('Open') : t('Closed')}
+            {statusText(status)}
```
[the remark this answers is not in the document]
Comment (chlebektomas) [reply project-base/storefront/components/Blocks/OpeningHours/OpeningStatus.tsx]: I like this approach.

File: project-base/storefront/components/Blocks/StoreList/StoreList.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { StoreListItem } from './StoreListItem';
+import { useEffect, useRef } from 'react';
+import { StoreOrPacketeryPoint } from 'utils/packetery/types';
+
+export const StoreList: FC<{ stores: StoreOrPacketeryPoint[]; selectedStoreUuid: string | null }> = ({
```
Comment (chlebektomas) [inline project-base/storefront/components/Blocks/StoreList/StoreList.tsx]: Please consider outsourcing the props in the same way as I suggested above.

File: project-base/storefront/components/Blocks/StoreList/StoreListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import { OpeningHours } from 'components/Blocks/OpeningHours/OpeningHours';
+import { OpeningStatus } from 'components/Blocks/OpeningHours/OpeningStatus';
+import { LinkButton } from 'components/Forms/Button/LinkButton';
+import { TypeOpeningHoursFragment } from 'graphql/requests/stores/fragments/OpeningHoursFragment.generated';
+import useTranslation from 'next-translate/useTranslation';
+import { Fragment, useEffect, useState } from 'react';
+import { StoreOrPacketeryPoint } from 'utils/packetery/types';
+import { twMergeCustom } from 'utils/twMerge';
+
+export const StoreListItem: FC<{ store: StoreOrPacketeryPoint; isSelected: boolean }> = ({ store, isSelected }) => {
```
Comment (chlebektomas) [inline project-base/storefront/components/Blocks/StoreList/StoreListItem.tsx]: Same outsourcing here, please.

File: project-base/storefront/components/Blocks/StoreList/StoreListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    const [isExpanded, setIsExpanded] = useState(false);
+    const { t } = useTranslation();
+
+    useEffect(() => {
+        setIsExpanded(isSelected);
+    }, [isSelected]);
+
+    const getCurrentOpeningHours = (openingHours: TypeOpeningHoursFragment) => {
```
Comment (chlebektomas) [inline project-base/storefront/components/Blocks/StoreList/StoreListItem.tsx]: Each time the `StoreListItem` component is rendered, this code is executed. I recommend moving it to a separate file or at least extracting it outside of this component.

File: project-base/storefront/components/Layout/SimpleLayout/SimpleLayout.tsx [hunk 6]
```diff
@@ -7,7 +7,7 @@ type SimpleLayoutProps = {
 
 export const SimpleLayout: FC<SimpleLayoutProps> = ({ heading, children, standardWidth }) => (
     <Webline>
-        <h1 className="text-center">{heading}</h1>
```
Comment (chlebektomas) [inline project-base/storefront/components/Layout/SimpleLayout/SimpleLayout.tsx:10]: `SimpleLayout` is used in three places. Please double-check if we want this to be global. If so, then it's fine.

**Review by chlebektomas — CHANGES_REQUESTED** [review]

File: project-base/storefront/components/Layout/SimpleLayout/SimpleLayout.tsx (hunk 6 above)
Comment (henzigo) [reply project-base/storefront/components/Layout/SimpleLayout/SimpleLayout.tsx:10]: We currently don't have design for those other two places. Let's merge it and after someone will design users consent and new password page he can look at it

File: project-base/storefront/components/Blocks/OpeningHours/OpeningStatus.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (henzigo) [reply project-base/storefront/components/Blocks/OpeningHours/OpeningStatus.tsx]: fixed

File: project-base/storefront/components/Blocks/OpeningHours/OpeningStatus.tsx (hunk 1 above)
Comment (henzigo) [reply project-base/storefront/components/Blocks/OpeningHours/OpeningStatus.tsx]: Fixed

File: project-base/storefront/components/Blocks/StoreList/StoreList.tsx (hunk 3 above)
Comment (henzigo) [reply project-base/storefront/components/Blocks/StoreList/StoreList.tsx]: Fixed

File: project-base/storefront/components/Blocks/StoreList/StoreListItem.tsx (hunk 4 above)
Comment (henzigo) [reply project-base/storefront/components/Blocks/StoreList/StoreListItem.tsx]: fixed

File: project-base/storefront/components/Blocks/StoreList/StoreListItem.tsx (hunk 5 above)
Comment (henzigo) [reply project-base/storefront/components/Blocks/StoreList/StoreListItem.tsx]: Moved to separated file.

**Review by chlebektomas — APPROVED** [review]

File: packages/frontend-api/src/Model/Store/OpeningHours/StoreOpeningHoursApiProvider.php [hunk 7]
```diff
@@ -12,6 +12,11 @@
 
 class StoreOpeningHoursApiProvider
 {
+    public const STATUS_OPEN = 'OPEN';
+    public const STATUS_CLOSED = 'CLOSED';
+    public const STATUS_OPEN_SOON = 'OPEN_SOON';
+    public const STATUS_CLOSED_SOON = 'CLOSED_SOON';
```
Comment (vitek-rostislav) [inline packages/frontend-api/src/Model/Store/OpeningHours/StoreOpeningHoursApiProvider.php]: missing strict type hint. Also, please consider moving the constants into the pseudo-enum class (see eg. https://github.com/shopsys/shopsys/blob/16.0/packages/framework/src/Model/Order/Item/OrderItemTypeEnum.php)

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]

File: packages/frontend-api/src/Model/Store/OpeningHours/StoreOpeningHoursApiProvider.php (hunk 7 above)
Comment (henzigo) [reply packages/frontend-api/src/Model/Store/OpeningHours/StoreOpeningHoursApiProvider.php]: fixed

**Review by vitek-rostislav — APPROVED** [review]

**Pull request merged**